File: Plugins/AM_MSGF_PlugIn/clsUniqueSeqInfo.vb
Option Strict On

Public Class clsUniqueSeqInfo
    Private mObsCount As Integer

    ''' <summary>
    ''' Observation count
    ''' </summary>
    ''' <returns></returns>
    ''' <remarks>Overridden in clsPSMInfo because it tracks specific PSMObservations</remarks>
    Public Overridable ReadOnly Property ObsCount As Integer
        Get
            Return mObsCount
        End Get
    End Property

    ''' <summary>
    ''' True if the C-terminus of the peptide is Lysine
    ''' </summary>
    Public Property CTermK As Boolean

    ''' <summary>
    ''' True if the C-terminus of the peptide is Arginine
    ''' </summary>
    Public Property CTermR As Boolean

    ''' <summary>
    ''' True if the peptide has an internal K or R that is not followed by P
    ''' </summary>
    ''' <returns></returns>
    Public Property MissedCleavage As Boolean

    ''' <summary>
    ''' True if the peptide is from a keratin protein
    ''' </summary>
    ''' <returns></returns>
    Public Property KeratinPeptide As Boolean

    ''' <summary>
    ''' True if the peptide is from a trypsin protein
    ''' </summary>
    ''' <returns></returns>
    Public Property TrypsinPeptide As Boolean

    ''' <summary>
    ''' True if the peptide is partially or fully tryptic
    ''' </summary>
    ''' <returns></returns>
    Public Property Tryptic As Boolean

    ''' <summary>
    ''' Constructor
    ''' </summary>
    Protected Sub New()
        Clear()
    End Sub

    Public Overridable Sub Clear()
        mObsCount = 0
        CTermK = False
        CTermR = False
        MissedCleavage = False
        KeratinPeptide = False
        TrypsinPeptide = False
        Tryptic = False
    End Sub

    Public Overridable Sub UpdateObservationCount(observationCount As Integer)
        mObsCount = observationCount
    End Sub
End Class

File: Plugins/AM_MSGF_PlugIn/MSGFResultsSummarizerDLL/clsUniqueSeqInfo.cs
namespace MSGFResultsSummarizer
{
    public class clsUniqueSeqInfo
    {
        private int mObsCount;

        /// <summary>
        /// Observation count
        /// </summary>
        /// <returns></returns>
        /// <remarks>Overridden in clsPSMInfo because it tracks specific PSMObservations</remarks>
        public virtual int ObsCount
        {
            get { return mObsCount; }
        }

        /// <summary>
        /// True if the C-terminus of the peptide is Lysine
        /// </summary>
        public bool CTermK { get; set; }

        /// <summary>
        /// True if the C-terminus of the peptide is Arginine
        /// </summary>
        public bool CTermR { get; set; }

        /// <summary>
        /// True if the peptide has an internal K or R that is not followed by P
        /// </summary>
        /// <returns></returns>
        public bool MissedCleavage { get; set; }

        /// <summary>
        /// True if the peptide is from a keratin protein
        /// </summary>
        /// <returns></returns>
        public bool KeratinPeptide { get; set; }

        /// <summary>
        /// True if the peptide is from a trypsin protein
        /// </summary>
        /// <returns></returns>
        public bool TrypsinPeptide { get; set; }

        /// <summary>
        /// True if the peptide is partially or fully tryptic
        /// </summary>
        /// <returns></returns>
        public bool Tryptic { get; set; }

        /// <summary>
        /// Constructor
        /// </summary>
        public clsUniqueSeqInfo()
        {
            Clear();
        }

        public virtual void Clear()
        {
            mObsCount = 0;
            CTermK = false;
            CTermR = false;
            MissedCleavage = false;
            KeratinPeptide = false;
            TrypsinPeptide = false;
            Tryptic = false;
        }

        public virtual void UpdateObservationCount(int observationCount)
        {
            mObsCount = observationCount;
        }
    }
}
